local-server/local-server-services/src/main/java/com/esri/samples/localserver/LocalServerServicesController.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2016 Esri.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.esri.samples.localserver;
+
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+
+import javafx.application.HostServices;
+import javafx.application.Platform;
+import javafx.event.ActionEvent;
+import javafx.event.EventTarget;
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.ListView;
+import javafx.scene.control.TextArea;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.localserver.LocalFeatureService;
+import com.esri.arcgisruntime.localserver.LocalGeoprocessingService;
+import com.esri.arcgisruntime.localserver.LocalMapService;
+import com.esri.arcgisruntime.localserver.LocalServer;
+import com.esri.arcgisruntime.localserver.LocalServerStatus;
+import com.esri.arcgisruntime.localserver.LocalService;
+import com.esri.arcgisruntime.util.ListenableList;
+
+public class LocalServerServicesController {
+
+  @FXML
+  private Button btnStartServer;
+  @FXML
+  private Button btnStopServer;
+  @FXML
+  private Button btnStartService;
+  @FXML
+  private Button btnStopService;
+  @FXML
+  private Button btnURL;
+  @FXML
+  private ComboBox<String> serviceOptions;
+  @FXML
+  private TextArea centerText;
+  @FXML
+  private ListView<String> runningServices;
+
+  private boolean isServerStopping;
+
+  private LocalServer server;
+  private ListenableList<LocalService> services;
+  private HostServices hostServices;
+
+  /**
+   * Handles starting and stopping a local server.
+   * 
+   * @param event the event that fired this action
+   */
+  @FXML
+  protected void handleStartStopLocalServer(ActionEvent event) {
+
+    // find out what fired this event
+    EventTarget eventTarget = event.getTarget();
+
+    // starting server
+    if (eventTarget.equals(btnStartServer)) {
+      // start local server
+      server = LocalServer.INSTANCE;
+      isServerStopping = false;
+      server.addStatusChangedListener(status -> {

```

Comment:
Label text could be set with a property on the server status if you choose to create observable properties.